File: JavaScript/Think-In-JS/Scope-In-JS/binarySearch.js
const arr = [3, 4, 5, 6, 7, 8, 9];
const length = arr.length;
const n = length - 1;
const data = +prompt("Enter your desire element");
let l = 0;
let r = n;

const binarySearch = (arr, low, high, data) => {
  while (low <= high) {
    let mid = low + (high-low) / 2;
    console.log(mid);
    if (data == arr[mid]) {
      return mid;
    } else if (data > arr[mid]) {
      return (low = mid + 1);
    } else {
      return (high = mid - 1);
    }
  }

  return -1;
};

const result = binarySearch(arr, l, r, data);

result == -1 ? console.log("Result not found") : console.log(`Your value position is ${result}`);

File: c/00 Practice/BinarySearch.c
#include <stdio.h>

int binarySearch(int arr[], int data, int low, int high)
{
    while(low <= high)
    {
        int mid = (low + high) / 2;
        if(data == arr[mid])
        {
            return mid;
        }
        else if(data > arr[mid])
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
    }

    return -1;
}

int main() {
    int arr[] = {5,7,9,12,15,20,24,27,33,40,50,65,72};
    int data, n, low, high, result;

    printf("Enter your desire element: ");
    scanf("%d", &data);

    n = sizeof(arr) / sizeof(arr[0]);
    low = 0;
    high = n - 1;

    result = binarySearch(arr, data, low, high);

    result == -1 ? printf("Desire element is not found") : printf("Desire element position is: %d", result);

    getch();
    return 0;
}
